vocabularies/Org.OData.Temporal.V1.xml

```diff
@@ -67,98 +67,82 @@
+          <Annotation Term="Core.LongDescription" String="A temporal object is a set of facts whose change over application and/or system time is tracked by the service. The entities of the annotated collection are time slices of potentially multiple temporal objects. A temporal object is uniquely identified within this set by the values of the specified object key properties. Object key properties follow the same rules as entity key properties. If no object key is specified, the annotated collection only contains time slices for a single one temporal object." />
         </Property>
         <Property Name="SupportedActions" Type="Collection(Temporal.QualifiedActionName)" Nullable="false">
           <Annotation Term="Core.Description" String="List of supported temporal actions" />
         </Property>
       </ComplexType>
 
-      <ComplexType Name="TimeDimensionType" Abstract="true">
-        <Property Name="SupportedQueries" Type="Temporal.QueryType" Nullable="false" />
+      <ComplexType Name="CollectionKindType" Abstract="true">
+        <Annotation Term="Core.Description" String="Kind of temporal collection" />
       </ComplexType>
 
-      <TypeDefinition Name="QualifiedActionName" UnderlyingType="Edm.String">
-        <Annotation Term="Core.Description" String="The qualified name of an action in scope." />
-        <Annotation Term="Core.LongDescription" String="*TODO: move this to Core, where we already have QualifiedTypeName and QualifiedTermName?*" />
-      </TypeDefinition>
+      <ComplexType Name="CollectionKindSnapshot">
+        <Annotation Term="Core.Description" String="Each OData entity corresponds to a temporal object and by default only shows data of the current time slice" />
+      </ComplexType>
+
+      <ComplexType Name="CollectionKindTimeSlice">
+        <Annotation Term="Core.Description" String="Each OData entity corresponds to a time slice and all time slices of a temporal object are part of the collection" />
+        <Property Name="TimeSliceStart" Type="Edm.PropertyPath" Nullable="false">
+          <Annotation Term="Core.Description" String="Property containing lower boundary of a time slice" />
+          <Annotation Term="Core.LongDescription" String="Must be present if time-period queries are supported" />
+        </Property>
+        <Property Name="TimeSliceEnd" Type="Edm.PropertyPath" Nullable="false">
+          <Annotation Term="Core.Description" String="Property containing upper boundary of a time slice" />
+          <Annotation Term="Core.LongDescription" String="Must be present if time-period queries are supported" />

```

Comment:
You write "must be present if ...", but it is not nullable. Contradiction?

Reply:
Leftover, now guaranteed by inheritance